hail/src/main/scala/is/hail/expr/ir/IR.scala

```diff
@@ -298,11 +298,8 @@ final case class ResultOp2(startIdx: Int, aggSigs: IndexedSeq[AggSignature]) ext
 
 final case class WriteAggs(startIdx: Int, path: IR, spec: CodecSpec, aggSigs: IndexedSeq[AggSignature]) extends IR
 final case class ReadAggs(startIdx: Int, path: IR, spec: CodecSpec, aggSigs: IndexedSeq[AggSignature]) extends IR
-
-//final case class WriteAggs(path: IR, spec: CodecSpec) extends IR
-//final case class ReadAggs(path: IR, spec: CodecSpec, aggSigs: Array[AggSignature]) extends IR
-//
-//final case class CombAggs(paths: IndexedSeq[IR], spec: CodecSpec, aggSigs: Array[AggSignature]) extends IR
+final case class SerializeAggs(startIdx: Int, serializedIdx: Int, spec: CodecSpec, aggSigs: IndexedSeq[AggSignature]) extends IR
+final case class DeserializeAggs(startIdx: Int, serializedIdx: Int, spec: CodecSpec, aggSigs: IndexedSeq[AggSignature]) extends IR
```

Comment:
What benefit do we get by putting (de)serialize in the IR, as opposed to just making the `FunctionWithAggRegion` trait know how to (de)serialize its aggregator state?

Reply:
I think it's difficult to be able to have the flexibility w.r.t the codec spec that we're using to serialize---that needs to be specified at compile time. We could say that we're only ever going to use one predetermined spec to serialize state, and then precompile methods to serialize all the states and have the ability to pass in an input/output stream and specify which aggregators to serialize.